doc/cascadia/Cascading-Default-Settings.md

```diff
@@ -213,31 +255,60 @@ generator runs, it can check if that profile source already has a profile
 associated with it, and do nothing (as to not create many duplicate "Ubuntu"
 profiles, for example).
 
+Additionally, each dynamic profile generator **must have a unique source guid**
+to associate with the profile. When a dynamic profile is generated, the source's
+guid will be added to the profile, to make sure the profile is correlated with
+the source it came from.
+
 When a dynamic profile generator runs, it will determine what new profiles need
 to be added to the user settings, and it can append those to the list of
 profiles. The generator will return some sort of result indicating that it wants
 a save operation to occur. The app will then save the `profiles.json` file,
 including these new profiles.
 
-If a dynamic profile generator has determined that a profile should no longer be
+We'll generate these dynamic profiles immediately after applying the defaults.
+When a generator runs, it'll be able to create unique profile GUIDs for each
+source it wants to generate a profile for. It'll attempt find any profiles that
+exist in the list of profiles with a matching GUID, and apply the settings to
+that profile if found, or it'll create a new profile to apply the settings to.

```

Comment:
this wording still suggests that the generator knows-about the user profile object and updates it.